openedx/core/djangoapps/waffle_utils/views.py

```diff
@@ -3,277 +3,124 @@
 """
 from collections import OrderedDict
 
-from django.conf import settings
-from edx_django_utils.monitoring import get_code_owner_from_module
 from edx_rest_framework_extensions.auth.jwt.authentication import JwtAuthentication
 from edx_rest_framework_extensions.permissions import IsStaff
-from edx_toggles.toggles import SettingDictToggle, SettingToggle
+from edx_toggles.toggles.state import ToggleStateReport, get_or_create_toggle_response
 from rest_framework import permissions, views
 from rest_framework.authentication import SessionAuthentication
 from rest_framework.response import Response
-from waffle.models import Flag, Switch
 
-from . import WaffleFlag, WaffleSwitch
 from .models import WaffleFlagCourseOverrideModel
 
 
-class ToggleStateView(views.APIView):
-    """
-    An endpoint for displaying the state of toggles in edx-platform.
-    """
-    authentication_classes = (JwtAuthentication, SessionAuthentication,)
-    permission_classes = (permissions.IsAuthenticated, IsStaff,)
-
-    def get(self, request):  # lint-amnesty, pylint: disable=missing-function-docstring
-        response = OrderedDict()
-        response['waffle_flags'] = _get_all_waffle_flags()
-        response['waffle_switches'] = _get_all_waffle_switches()
-        response['django_settings'] = _get_settings_state()
-        return Response(response)
-
-
-def _get_all_waffle_switches():
+class CourseOverrideToggleStateReport(ToggleStateReport):
     """
-    Gets all waffle switches and their state.
+    Override some of the methods from ToggleStateReport to expose toggles from WaffleFlagCourseOverrideModel objects.
     """
-    switches_dict = {}
-    _add_waffle_switch_instances(switches_dict)
-    _add_waffle_switch_state(switches_dict)
-    _add_waffle_switch_computed_status(switches_dict)
-    switch_list = list(switches_dict.values())
-    switch_list.sort(key=lambda toggle: toggle['name'])
-    return switch_list
 
+    def add_waffle_flag_instances(self, flags_dict):
+        """
+        Append objects from WaffleFlagCourseOverrideModel.
+        """
+        super().add_waffle_flag_state(flags_dict)
+        _add_waffle_flag_course_override_state(flags_dict)
 
-def _add_waffle_switch_instances(switches_dict):
-    """
-    Add details from waffle switch instances, like code_owner.
-    """
-    waffle_switch_instances = WaffleSwitch.get_instances()
-    for switch_instance in waffle_switch_instances:
-        switch = _get_or_create_toggle_response(switches_dict, switch_instance.name)
-        _add_toggle_instance_details(switch, switch_instance)
+    def get_waffle_flag_computed_status(self, flag):
+        """
+        Produce correct "computed_status" values for WaffleFlagCourseOverrideModel instances.
```

Comment:
Should we enhance test coverage? Note: adding the test before the refactor would ensure the refactor isn’t changing anything.

Reply:
Test coverage could certainly be improved -- there is currently just one test that checks the `computed_status` value.